src/main/scala/LSWidgets.scala

```diff
@@ -180,95 +168,110 @@ class Relationship(val key: WidgetKey, val state: State, val ws: GUIWorkspace) e
-  add(new JLabel("ask"), "align right")
+  add(new JLabel(I18N.get("relationship.agentset")), "align right")
   val agentSelector: XWComboBox = new XWComboBox(() => updateInState(kind.selectedAgentReporterProperty))
   add(agentSelector, "grow, wrap")
 
   val agentArgumentPanel = new JPanel()
   add(agentArgumentPanel, "grow, span, wrap")
 
-  add(new JLabel("to do"), "align right")
+  val agentsetArgumentPanel = new LSArgumentSelector(() => updateInState(kind.selectedAgentsetArguments), ws)
+  add(agentsetArgumentPanel, "grow, span, wrap")
+
+  add(new JLabel(I18N.get("relationship.commands")), "align right")
   val procedureSelector: XWComboBox = new XWComboBox(() => updateInState(kind.selectedProcedureProperty))
-  procedureSelector.onItemStateChanged { event ⇒
-    if (event.getStateChange == ItemEvent.SELECTED)
-      updateInState(kind.selectedProcedureProperty)
-  }
   add(procedureSelector, "grow, wrap")
 
-  val procedureArgumentPanel = new JPanel()
+  val procedureArgumentPanel = new LSArgumentSelector(() => updateInState(kind.selectedProcedureArguments), ws)
   add(procedureArgumentPanel, "grow, span, wrap")
 
   val buttonPanel = new JPanel()
-  val saveButton = new JButton("save")
+  val saveButton = makeButton("relationship.save", tryCompilation(ws, owner, () => saveCommand))
   buttonPanel.add(saveButton)
-  saveButton.onActionPerformed(_ =>
-    try ws.evaluateCommands(owner, saveCommand, ws.world.observers, waitForCompletion = false)
-    catch { case e: CompilerException => ws.warningMessage(e.getMessage) }
-  )
 
-  val deleteButton = new JButton("delete")
+  val deleteButton = makeButton("relationship.delete", tryCompilation(ws, owner, () => deleteCommand))
   buttonPanel.add(deleteButton)
-  deleteButton.onActionPerformed(_ =>
-    try ws.evaluateCommands(owner, deleteCommand, ws.world.observers, waitForCompletion = false)
-    catch { case e: CompilerException => ws.warningMessage(e.getMessage) }
-  )
+
+  val runButton = makeButton("relationship.run", tryCompilation(ws, owner, () => runCommand))
+  buttonPanel.add(runButton)
 
   add(buttonPanel, "grow, span")
 
-  def selectedProcedureArguments =
-    procedureArguments.map {
+  def selectedAgentsetArguments: LogoList = agentsetArgumentPanel.selectedArguments
+  def selectedAgentsetArguments_=(args: LogoList): Unit = agentsetArgumentPanel.selectedArguments = args
+
+  def availableAgentsetArguments: LogoList = agentsetArgumentPanel.availableArguments
+  def availableAgentsetArguments_=(args: LogoList) = agentsetArgumentPanel.availableArguments = args
+
+  def selectedProcedureArguments: LogoList = procedureArgumentPanel.selectedArguments
+  def selectedProcedureArguments_=(args: LogoList): Unit = procedureArgumentPanel.selectedArguments = args
+
+  def availableProcedureArguments: LogoList = procedureArgumentPanel.availableArguments
+  def availableProcedureArguments_=(args: LogoList) = procedureArgumentPanel.availableArguments = args
```

Comment:
You don't need proxy accessors anymore.  Just toss them and directly access the `LSArgumentSelector` accessors in the properties.